fix reference to resolve_names
